docs/api.md

```diff
@@ -1273,7 +1273,7 @@ or updates existing device details for this user/session
 If no device `id` is specified,
 both `name` and `type` must be provided.
 If a device `id` is specified,
-at least one of `name`, `type`, `pushCallback` and `pushPublicKey`
+at least one of `name`, `type`, `pushCallback`, `pushPublicKey` and `pushAuthKey`
```

Comment:
I like the 2nd solution, ideally we want the 3 fields, I'll see if it is possible to validate their existence at the same time